extensions/data-plane/data-plane-kafka/src/main/java/org/eclipse/edc/dataplane/kafka/pipeline/KafkaDataSink.java

```diff
@@ -0,0 +1,103 @@
+/*
+ *  Copyright (c) 2023 Amadeus
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Amadeus - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.dataplane.kafka.pipeline;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.eclipse.edc.connector.dataplane.spi.pipeline.DataSource;
+import org.eclipse.edc.connector.dataplane.util.sink.ParallelSink;
+import org.eclipse.edc.spi.response.StatusResult;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+
+import static org.eclipse.edc.spi.response.ResponseStatus.FATAL_ERROR;
+
+class KafkaDataSink extends ParallelSink implements Closeable {
+
+    private Properties producerProperties;
+    private String topic;
+    private Producer<String, byte[]> producer;
+
+    private KafkaDataSink() {
+    }
+
+    public void start() {
+        monitor.debug(String.format("start producer with properties: %s, and producing on topic: %s", producerProperties, topic));
+        producer = new KafkaProducer<>(producerProperties);
+    }
+
+    @Override
+    public void close() {
+        if (producer != null) {
+            producer.close();
+        }
+    }
+
+    @Override
+    protected StatusResult<Void> transferParts(List<DataSource.Part> parts) {
+        return parts.stream()
+                .map(this::publishPart)
+                .filter(StatusResult::fatalError)
+                .findFirst()
+                .orElse(StatusResult.success());
+    }
+
+    private StatusResult<Void> publishPart(DataSource.Part part) {
+        try (var is = part.openStream()) {
+            // asynchronous publishing
+            producer.send(new ProducerRecord<>(topic, null, is.readAllBytes()), (metadata, exception) -> {
+                if (exception != null) {
```

Comment:
swallowing the exception seems kinda ... smelly. We should at least log it, i.e. passing it to the `warning()` method. Possibly we should even return `StatusResult.failure(...)` in that case